python/cdp/to_network_scoped_evm_server_account.py

```diff
@@ -0,0 +1,331 @@
+        if is_method_supported_on_network("swap", self._network):
+            self._supported_methods["swap"] = self._network_scoped_swap
+
+    def __getattr__(self, name: str) -> Any:
+        """Dynamically access supported methods and account properties, or raise AttributeError if not found."""
+        if name in self._supported_methods:
+            return self._supported_methods[name]
+        # Expose account properties
+        if name == "network":
+            return self._network
+        if name == "rpc_url":
+            return self._rpc_url
+        if name == "address":
+            return self._evm_server_account.address
+        if name == "name":
+            return getattr(self._evm_server_account, "name", None)
+        if name == "type":
+            return "evm-server"
+        if name == "policies":
+            return getattr(self._evm_server_account, "policies", None)
+        # Expose signing methods
+        if name == "sign":
+            return getattr(self._evm_server_account, "sign", None)
+        if name == "sign_message":
+            return getattr(self._evm_server_account, "sign_message", None)
+        if name == "sign_transaction":
+            return getattr(self._evm_server_account, "sign_transaction", None)
+        if name == "sign_typed_data":
+            return getattr(self._evm_server_account, "sign_typed_data", None)
+        raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
+
+    async def _network_scoped_send_transaction(
+        self,
+        transaction: str | dict[str, Any],
+        idempotency_key: str | None = None,
+    ) -> str:
+        """Send a transaction using the API for managed networks, or directly via web3.py for custom RPC on non-managed networks.
+
+        If a private key is provided, sign and send the transaction dict. Otherwise, require a signed tx hex string.
+        """
+        if self._web3:
+            if self._signer and isinstance(transaction, dict):
+                # Fill in from, sign, and send
+                tx = dict(transaction)
+                if "from" not in tx:
+                    tx["from"] = self._signer.address
+                if "nonce" not in tx:
+                    tx["nonce"] = self._web3.eth.get_transaction_count(self._signer.address)
+                if "gas" not in tx:
+                    tx["gas"] = self._web3.eth.estimate_gas(tx)
+                if "gasPrice" not in tx and "maxFeePerGas" not in tx:
+                    tx["gasPrice"] = self._web3.eth.gas_price
+                signed = self._signer.sign_transaction(tx)
+                tx_hash = self._web3.eth.send_raw_transaction(signed.rawTransaction)
+                return self._web3.toHex(tx_hash)
+            elif isinstance(transaction, str):
+                # If transaction is a raw signed tx hex string
+                tx_hash = self._web3.eth.send_raw_transaction(transaction)
```

Comment:
if we use send_transaction instead of send_raw_transaction, i believe web3 will do the gas estimation for us